src/darwin/Framework/CHIP/MTRMetrics.h

```diff
@@ -20,6 +20,31 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ * Representation of metric data corresponding to a metric event.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRMetricData : NSObject
+
+/**
+ * Value for the metric data. The value may be nil depending on the event emitted.
+ */
+@property (nonatomic, nullable, readonly, copy) NSNumber * value;
+
+/**
+ * Error code for the metric data. This value when valid, holds the error code value
+ * of the operation associated with the event.
```

Comment:
It would be good to document what value space these error codes come from.  Are these CHIP_ERROR values?  Something else?

They look like CHIP_ERROR values to me, which is hard to document in our public API....  I wonder whether we should convert these to NSError instead of just reporting the raw codes, but that conversion can be lossy for error types we don't know about.

If this stays as-is, it needs at least _some_ documentation that would allow consumers to interpret these values.

Reply:
This is subject to the metric and could be a non-CHIP error as well. Hence I left it open ended.